archunit/src/main/java/com/tngtech/archunit/core/domain/properties/CanBeAnnotated.java

```diff
@@ -157,7 +159,24 @@ public static boolean isMetaAnnotatedWith(
                 DescribedPredicate<? super JavaAnnotation<?>> predicate) {
 
             for (JavaAnnotation<?> annotation : annotations) {
-                if (annotation.getRawType().isAnnotatedWith(predicate) || annotation.getRawType().isMetaAnnotatedWith(predicate)) {
+                if (isMetaAnnotatedWith(annotation, predicate, new HashSet<String>())) {
```

Comment:
I think `annotation.getRawType().isAnnotatedWith(predicate)` does the same as now within `isMetaAnnotatedWith(..)`. Because `annotation.getRawType().isAnnotatedWith(predicate)` will loop over all annotations on the raw type of the annotation and look if the predicate matches. Now if we dive into `isMetaAnnotatedWith(..)` we will end in a loop over `annotation.getRawType().getAnnotations()` and then directly check the predicate `predicate.apply(metaAnnotation)`.
So from my understanding the behavior should be the same before and after and both times only have checked for "pure" meta annotations (I'm pretty sure about it, because I was pondering if this is even useful behavior :joy: Because for most cases you're interested if the annotation is direct or meta-annotated, but I cannot think of any case where you would really care about "pure" meta-annotations).
Or am I seeing this wrong? :thinking: But originally we put the `predicate` into `isAnnotatedWith(..)`, while now we do a direct `predicate.apply(metaAnnotation)` inside of `isMetaAnnotatedWith(..)`.